base/src/commonMain/kotlin/utils/Decimal.kt

```diff
@@ -192,17 +336,36 @@ sealed class RoundingMode {
  * @param roundingMode The [RoundingMode] to apply when scaling.
  * @return A [Decimal] rounded to [scale] digits after the decimal point.
  */
-expect fun Decimal.round(scale: Int, roundingMode: RoundingMode = RoundingMode.RoundHalfEven): Decimal
+fun Decimal.round(scale: Int, roundingMode: RoundingMode = RoundingMode.RoundHalfEven): Decimal = when (this) {
+    is Decimal.Finite -> Decimal.Finite(finiteDecimal.round(scale, roundingMode))
+    else -> this
+}
+internal expect fun FiniteDecimal.round(scale: Int, roundingMode: RoundingMode = RoundingMode.RoundHalfEven): FiniteDecimal
 
 /**
  * Converts a [Number] to a [Decimal]
  */
-expect fun Number.toDecimal(): Decimal
+fun Number.toDecimal(): Decimal = when {
+    toDouble().isFinite() -> Decimal.Finite(toFiniteDecimal())
+    toDouble().isNaN() -> Decimal.NaN
+    toDouble() == Double.POSITIVE_INFINITY -> Decimal.PositiveInfinity
+    else -> Decimal.NegativeInfinity
+}
 
 /**
  * Converts a String to a [Decimal]
  */
-expect fun String.toDecimal(): Decimal
+fun String.toDecimal(): Decimal = when (this) {
+    Double.NaN.toString() -> Decimal.NaN
+    Double.POSITIVE_INFINITY.toString(),
+    '\u221E'.toString() -> Decimal.PositiveInfinity
+    Double.NEGATIVE_INFINITY.toString(),
+    "-${'\u221E'}" -> Decimal.NegativeInfinity
+    else -> toFiniteDecimal()?.let { Decimal.Finite(it) } ?: Decimal.NaN
```

Comment:
This is doesn't correspond existing standard library API, where it throws on try to decode string into int/float.
Because string "foo" is not a type of double which contains not a number.
Logically it should throw, or at least return null in such cases

Reply:
NSDecimalNumber returns NSDecimalNumber.notANumber for these cases. I figured Id do the same